File: cpp/0122买卖股票的最佳时机Ⅱ.cpp
#include<vector>

using namespace std;


//贪心
int maxProfit(vector<int> &prices)
{
    //只要在两天之内有利润就进行一次买入卖出操作
    //无利润则不买

    int res = 0;

    for (int i = 1;i<prices.size();++i)
    {
        int intrests = prices[i] - prices[i - 1];
        res = intrests > 0 ? res + intrests : res;
    }

    return res;
}

File: csharp/0122买卖股票的最佳时机Ⅱ.cs

public partial class Solution
{
    //贪心
    public int MaxProfit1(int[] prices)
    {
        int res = 0;

        for (int i = 1; i < prices.Length; i++)
        {
            res += Math.Max(0, prices[i] - prices[i-1]);
        }
        return res;
    }

    //dp
    public int MaxProfit2(int[] prices)
    {
        int[] dp = new int[2];
        dp[0] = -prices[0];

        for (int i = 1; i < prices.Length; i++)
        {
            dp[0] = dp[0]>dp[1] - prices[i]?dp[0]:dp[1] - prices[i];
            dp[1] = dp[1] > dp[0] + prices[i] ? dp[1] : dp[0] + prices[i];
        }
        return dp[1];
    }
}